### ISSUE TITLE
Failures on send_feedback_raw path when using proto

### ISSUE THREAD

**Issue by matthewphsmith** [SeldonIO/seldon-core#2606]
## Describe the bug
Hello, we recently upgraded our version of seldon-core (1.4.0) and have begun experiencing the following failure on our `send_feedback` path. After our custom `send_feedback_raw` method returns a `SeldonMessage` pb object as expected, it enters the handle_raw_custom_metrics method which incorrectly treats the object as a dict.  It appears to me that this [line](https://github.com/SeldonIO/seldon-core/blob/b59d2a32762460e6d1ad59cbce495ee1ff4078a3/python/seldon_core/seldon_methods.py#L179) might be a typo and the root cause. The type hint indicates the input proto should be a `Feedback` pb object so the `isinstance` check will always be false and thus cause the metric method to treat the response as a dict.
```
File "/code/venvs/venv/lib/python3.6/site-packages/seldon_core/seldon_methods.py", line 196, in send_feedback
response, seldon_metrics, is_proto, FEEDBACK_METRIC_METHOD_TAG
File "/code/venvs/venv/lib/python3.6/site-packages/seldon_core/seldon_methods.py", line 63, in handle_raw_custom_metrics
metrics = msg.get("meta", {}).get("metrics", [])
```

## To reproduce

1. Write a model with a custom `send_feedback_raw` method which returns a `SeldonMessage`
2. Build image with seldon-core=1.4.0
3. Call send_feedback REST endpoint

## Expected behaviour
I would expect `is_proto=isinstance(request, prediction_pb2.Feedback)` thus resulting in proper handling of the proto output from `send_feedback_raw` in `handle_raw_custom_metrics`

**Comment by matthewphsmith** [comment]
As a newbie, if someone can confirm my reading of the code isn't deeply misguided, I'd be happy to open a PR for this.

**Comment by cliveseldon** [comment]
Thnaks @matthewphsmith 

@RafalSkolasinski any thoughts?

**Comment by RafalSkolasinski** [comment]
@matthewphsmith could you provide what is the `Exception` thrown in the python stack as well as example on what you return from the `send_feedback_raw`?

**Comment by matthewphsmith** [comment]
@RafalSkolasinski On phone, but the error was that object did not have a `get` attribute. I return an empty `SeldonMessage()`.

**Comment by RafalSkolasinski** [comment]
@matthewphsmith Thanks! If you could later (no rush) share code of the model that you wrap that would be great (just to make sure that we look at the same issue).

**Comment by matthewphsmith** [comment]
We wrote an entirely custom model object that implements the internal API methods like `send_feedback_raw` and `predict_raw`. It's difficult for me to share the code in all it's complexity, but here is a copy of the `send_feedback_raw` method:

```
    def send_feedback_raw(self, feedback: Feedback) -> SeldonMessage:
        metric_tags = self._get_base_stats_tags()
        failure_phase_tag = "failure_phase"
        failure_metric = "send_feedback_failure_cnt"

        with _stats.timer("send_feedback_raw_ms", tags=metric_tags):
            with _stats.timer("send_feedback_deserialize_ms", tags=metric_tags):
                try:
                    compressed_data = base64.b64decode(feedback.request.strData)
                    input_data = gzip.decompress(compressed_data).decode()
                    datasets: Datasets = deserialize_json_datasets(input_data)
                except DeserializationError as e:
                    logger.exception(
                        "Error deserializing Dataset from SeldonMessage strData"
                    )
                    metric_tags[failure_phase_tag] = "deserialize"
                    _stats.incr(failure_metric, tags=metric_tags)
                    raise SeldonMicroserviceException(
                        "Error deserializing inputs.", status_code=400, reason=str(e)
                    )

            with _stats.timer("send_feedback_user_ms", tags=metric_tags):
                try:
                    self.send_feedback_with_backup(datasets)
                except Exception as e:
                    logger.exception("Error updating weights", kv={"reason": str(e)})
                    metric_tags[failure_phase_tag] = "user_send_feedback"
                    _stats.incr(failure_metric, tags=metric_tags)
                    raise SeldonMicroserviceException(
                        "Error updating weights.", status_code=500, reason=str(e)
                    )

            _stats.incr("send_feedback_success_cnt", tags=metric_tags)
            return SeldonMessage()
```

**Comment by RafalSkolasinski** [comment]
@matthewphsmith https://github.com/SeldonIO/seldon-core/pull/2639 that was just merged may hopefully resolved this - would you be so kind to try now? You could use `seldonio/seldon-core-s2i-python37:1.5.0-dev` dev image.

**Comment by matthewphsmith** [comment]
@RafalSkolasinski, thank you for the follow-up! That change does prevent the crash in our case because it sidesteps attempting to pull metrics from our response. However, I don't think it is complete in terms of functionality of the library. For instance, if I had returned `metrics` in my response metadata, it would not be processed correctly by the library--the metrics will be ignored.

I still suspect this is the root cause: https://github.com/axsaucedo/seldon-core/blob/e555400ba1637cdbd640da64d5ac79f3cfc11a17/python/seldon_core/seldon_methods.py#L180

The type signature of the method expects a `prediction_pb2.Feedback` object. The `isinstance` check at the line linked would thus always be `false` as it is checking for an instance of a `prediction_pb2.SeldonMessage`. This informs the `handle_raw_custom_metrics' method that the input object is not a proto. However, since our method adheres to the expected type signature and returns a `prediction_pb2.SeldonMessage`, I would want the `handle_raw_custom_metrics` method to handle my response as a proto, convert to json, and inspect the metadata for metrics.

Thus I suggest replacing the aforementioned line with `is_proto = isinstance(request, prediction_pb2.Feedback)`.

**Comment by cliveseldon** [comment]
@matthewphsmith I agree this check seems wrong. Have you tried fixing that one line and testing your use case?

**Comment by matthewphsmith** [comment]
@cliveseldon sorry for the slow reply, yes, adjusting that one line does work for my use-case. I can open a PR if you'd like!

**Comment by cliveseldon** [comment]
Yes please!

**Comment by matthewphsmith** [comment]
@cliveseldon sorry again for yet another very slow reply, but I have opened a very minor PR which I believe rectifies the issue fully.

Please note: This [change](https://github.com/axsaucedo/seldon-core/commit/f8d87bf6974de28f9b7d5095aa7d4ed42e8b48c9#diff-8d24a000e0490afd1da58a13db35ef7635a4dc49bdc72d12cceb465d6831c9dfL62) rectifies the crash issue mentioned earlier in this issue. However, there is still a gap (which the attached PR fixes) in that the `handle_custom_metrics` method will not pull custom metrics from the response from `send_feedback` because `is_proto` is erroneously `False` and the object returned is never a dict.
